### DISCUSSION TITLE
How is the official support of null as input value in mutations?

### DISCUSSION THREAD

**Discussion by pcMarcos** [rmosolgo/graphql-ruby#3661] [Q&A] [answered]
hey everyone, 

I'm currently using GraphQL-ruby in one of my projects and found something curious. 

If I use `null` as a input value in mutations, it deletes the attribute in my database. 

Like this:

`{
  "id": 1,
  "name": "John",
  "hobby": null
}
`
It deletes the value in "hobby" for this user.

Whats the official support of this keyword in your ruby implementation of GraphQL? I couldn't find any mention of it in the documentation.

**Comment by rmosolgo** [comment, marked as answer]
Hi! Yes, in GraphQL, `null` means "the value here is `null`", not "there is no value here." Here's the spec which I've tried to implement: 

https://spec.graphql.org/draft/#sec-Null-Value

> For example, a mutation representing deleting a field...

From that example, it seems to me that GraphQL-Ruby matches the spec. By that reading, for`"hobby"` to remain unchanged, don't pass `"hobby": ...` at all. Does that sound right to you?

**Reply by pcMarcos** [reply]
great! it sounds fine! thank you

**Comment by verdihatorou** [comment]
Helow
